desktop/src/main/java/bisq/desktop/primary/main/content/social/chat/components/PublicChannelSelection.java

```diff
@@ -41,130 +42,23 @@
-            selectedChannelPin = FxBindings.subscribe(chatService.getPersistableStore().getSelectedChannel(),
-                    channel -> {
-                        if (channel instanceof PublicChannel publicChannel) {
-                            model.selectedChannel.set(new ListItem(publicChannel));
-                        }
-                    });
-            channelsPin = FxBindings.<PublicChannel, ListItem>bind(model.channels)
-                    .map(ListItem::new)
-                    .to(chatService.getPersistableStore().getPublicChannels());
-
-            if (!model.channels.isEmpty()) {
-                chatService.selectChannel(model.channels.get(0).channel);
-            }
-        }
-
-        @Override
-        public void onViewDetached() {
-            selectedChannelPin.unbind();
-            channelsPin.unbind();
-        }
-
-        private void onSelected(ListItem selectedItem) {
-            if (selectedItem == null) return;
-            chatService.selectChannel(selectedItem.channel);
-        }
-    }
-
-    private static class Model implements bisq.desktop.common.view.Model {
-        ObjectProperty<ListItem> selectedChannel = new SimpleObjectProperty<>();
-        ObservableList<ListItem> channels = FXCollections.observableArrayList();
-
-        private Model() {
-        }
-    }
-
-
-    @Slf4j
-    public static class View extends bisq.desktop.common.view.View<VBox, Model, Controller> {
-        private final ListView<ListItem> listView;
-        private Subscription subscription;
-
-        private View(Model model, Controller controller) {
-            super(new VBox(), model, controller);
-            root.setSpacing(10);
-
-            BisqLabel headline = new BisqLabel(Res.get("social.publicChannels"));
-            headline.setPadding(new Insets(0, 0, 0, 10));
-            headline.setStyle("-fx-text-fill: -bs-color-green-5; -fx-font-size: 1.4em");
-
-            listView = new ListView<>();
-            listView.setItems(model.channels);
-            listView.setFocusTraversable(false);
-            listView.setCellFactory(new Callback<>() {
-                @Override
-                public ListCell<ListItem> call(ListView<ListItem> list) {
-                    return new ListCell<>() {
-                        final BisqLabel label = new BisqLabel();
-
-                        @Override
-                        public void updateItem(final ListItem item, boolean empty) {
-                            super.updateItem(item, empty);
-                            if (item != null && !empty) {
-                                label.setText(item.channelName);
-                                setGraphic(label);
-                            } else {
-                                setGraphic(null);
-                            }
-                        }
-                    };
+        super();
+        controller = new ChannelSelection.Controller(chatService, Res.get("social.publicChannels")) {
```

Comment:
We could pass only the display string in the super call and initiate the controller in the base class and make it final.